embedded-compositor/ScreenShotInterface.cpp

```diff
@@ -0,0 +1,78 @@
+/*!
+ * @file   ScreenShotInterface.cpp
+ *
+ * @date   10.07.2024
+ *
+ * @author Christian Happ
+ *
+*/
+
+#include "ScreenShotInterface.hpp"
+#include "screenshot_adaptor.h"
+
+#include <QImage>
+#include <QDateTime>
+#include <QDebug>
+#include <QQuickWindow>
+
+
+ScreenShotInterface::ScreenShotInterface(QQmlApplicationEngine *engine, QObject *parent)
+    : DBusInterface(QStringLiteral("/screenshot"), parent),
+    m_pEngine(engine),
+    m_ImageQuality(50),
+    m_ImageFormat("PNG"),
+    m_FileExtension(".png"),
+    m_BaseFileName("screenshot_")
+{
+    new ScreenshotAdaptor(this);
+
+    // register object by dbus service name
+    componentComplete();
+}
+
+QString ScreenShotInterface::getISODate() const
+{
+    QDateTime dateTime;
+    dateTime = QDateTime::currentDateTime();
+    QString strDateTime = dateTime.toString(Qt::ISODate);
+
+    return strDateTime;
+}
+
+QString ScreenShotInterface::generateFilename() const
+{
+    QString concateFileName = m_BaseFileName + getISODate() + m_FileExtension;
+
+    return concateFileName;
+}
+
+QString ScreenShotInterface::ScreenShot(const QString storePath)
+{
+    QString fileSavePath = {};
```

Comment:
```suggestion
    QString fileSavePath;
```
don't initialize the String objects, the default constructor is the better way

Reply:
done